src/server/game/Entities/Creature/Creature.h

```diff
@@ -662,8 +662,8 @@ class Creature : public Unit, public GridObject<Creature>, public MovableMapObje
             return m_charmInfo->GetCharmSpell(pos)->GetAction();
     }
 
-    void SetCannotReachTarget(bool cannotReach) { if (cannotReach == m_cannotReachTarget) return; m_cannotReachTarget = cannotReach; m_cannotReachTimer = 0; }
-    [[nodiscard]] bool CanNotReachTarget() const { return m_cannotReachTarget; }
+    void SetCannotReachTarget(bool cannotReach);
+    [[nodiscard]] bool CanNotReachTarget() const { return (GetMap() && GetMap()->IsRaid()) ? false : m_cannotReachTarget; }
```

Comment:
The condition GetMap() && GetMap()->IsRaid() is redundant, you're already checking the IsRaid stuff within the previous code

Reply:
Please optimize/clean your code